### PR TITLE
ADD Progression (enemies get faster, amount over time)

### PR DESCRIPTION
**Pull request by DaXzZz**
Test ๆ

### CODE REVIEW

File: Assets/Scenes/Scripts/GameManager.cs [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
-        if (enemySpawnTime >= 1f) {
+        if (enemySpawnTime >= enemySpawnInterval) {
             enemies.Add(Instantiate(enemyPrefab, player.transform.position + (Vector3)Random.insideUnitCircle.normalized * 10f, Quaternion.identity));
             enemySpawnTime = 0;
         }
-        foreach (var en in enemies) {
-            if (en != null) {
-                // TODO: Enemy move speed on each type
-                en.GetComponent<Rigidbody2D>().MovePosition(en.transform.position + (Vector3)(player.transform.position - en.transform.position).normalized * Time.deltaTime * 2f);
+
```
Comment (ikkyuuq) [inline Assets/Scenes/Scripts/GameManager.cs]: Remove all empty line, space as much as possible

File: Assets/Scenes/Scripts/GameManager.cs [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
+public class GameManager : MonoBehaviour
+{
     public GameObject playerPrefab, enemyPrefab;
-    public float playerMoveSpeed = 5f, enemySpawnTime = 0f;
+    public float playerMoveSpeed = 5f, enemySpawnTime = 0f ,  enemySpawnInterval = 1f , enemyMoveSpeed = 3f , progressionInterval = 5f , spawnRateIncrease = 0.1f , speedIncrease = 0.5f , progressionTimer = 0f;
     private GameObject player;
     private List<GameObject> enemies = new List<GameObject>(), projectiles = new List<GameObject>();
-    void Start(){
+    void Start()
+    {
```
Comment (ikkyuuq) [inline Assets/Scenes/Scripts/GameManager.cs]: void Start() {

File: Assets/Scenes/Scripts/GameManager.cs [hunk 3]
```diff
@@ -1,33 +1,46 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
-public class GameManager : MonoBehaviour{
+public class GameManager : MonoBehaviour
+{
```
Comment (ikkyuuq) [inline Assets/Scenes/Scripts/GameManager.cs]: public class GameManager : MonoBehaviour {

File: Assets/Scenes/Scripts/GameManager.cs [hunk 4]
```diff
@@ -1,33 +1,46 @@
...
-    void Update(){
+    void Update() {
         PlayerMovement();
         EnemyManager();
-        //so on... jai dee laew na krub :)
+        HandleProgression();
     }
-    void PlayerMovement() {
+    void PlayerMovement()
+    {
```
Comment (ikkyuuq) [inline Assets/Scenes/Scripts/GameManager.cs]: void PlayerMovement() {

**Review by ikkyuuq — CHANGES_REQUESTED** [review]
- [x] remove all empty line 
- [x] refactor :4
- [x] refactor :10
- [x] refactor :19

**Review by DaXzZz — COMMENTED** [review]
Remove Line เเล้วเด้อ

**Comment by DaXzZz** [comment]
เริ่มดึก เริ่มหลอน

**Review by ikkyuuq — APPROVED** [review]
nice work! krub :+1:

**Pull request merged**